src/proof/correct_key.rs

```diff
@@ -62,11 +63,11 @@ pub trait ProveCorrectKey<EK, DK> {
     fn prove(dk: &DK, challenge: &Challenge) -> Result<CorrectKeyProof, ProofError>;
 
     /// Verify proof.
-    fn verify(proof: &CorrectKeyProof, aid: &VerificationAid) -> Result<(), ProofError>;
+    fn verify_correctKey(proof: &CorrectKeyProof, aid: &VerificationAid) -> Result<(), ProofError>;
 }
 
-fn compute_digest<IT>(values: IT) -> BigInt
-where  IT: Iterator, IT::Item: Borrow<BigInt>
+pub fn compute_digest<IT>(values: IT) -> BigInt
```

Comment:
Can we extract this in a utility (I know we have this as a task, can we at least add a comment) or a TODO